src/java_bytecode/java_bytecode_convert_method_class.h

```diff
@@ -81,6 +83,10 @@ class java_bytecode_convert_methodt:public messaget
   /// Initialized in `convert`.
   unsigned slots_for_parameters;
 
+  /// True if the driver program will provide bodies for all stub functions;
+  /// will be used to annotate stubs created upon discovery of a callsite.
+  bool driver_program_provides_stubs;
```

Comment:
Is there a use case for this?

Reply:
Test-gen acts this way: stub functions are given a body shortly after creation.

Reply:
Trying to understand how this will fit with the discussions around https://github.com/diffblue/cbmc/pull/1585 in the end...

Reply:
At the moment if you (the driver program) have a non-default plan for stubs, you should set this parameter. For example, test-gen derives from `java_bytecode_languaget` in order to make similar customisations, and will set this once this change hits CBMC.

Later I would like for stubbing to take place via a standard callback provided by languaget, such that the driver program supplies an implementation there instead of customising the language frontend. There could be standard implementations (replace with assume-false, add a Java-style nondet-object-graph stub, ...), or the driver could do something cleverer.

Reply:
I have no idea about urgencies, but really I'd want to see this addressed before merging this PR (knowing that this blocks a series of other PRs). My problem with the chosen approach is that `function_is_stub` is a piece of configuration information, yet is never documented as such in, e.g., the symbol table. That is, the semantics of the program depend on `function_is_stub`, but there is no way to recover from one choice or the other.

There is an ugly option of delaying things until symbolic execution, which would figure out that no function definition is present, and thus create an assignment to the return-value variable instead (where the communication that a return value is required is implicit via the existence of the return-value symbol). The advantage is that goto programs can be changed up to the very last moment.

The proposed "standard callback" solution seems preferable, but is possibly more work. Indeed the discussion in #1585 should be driven to a conclusion.

Reply:
So this actually isn't quite the same as the please-provide-a-stub callback, rather the question is, of a function that has *not yet* been loaded, will a body ever be provided? Therefore I propose for this PR I'll implement `function_will_have_body`, a callback provided by the driver program (jbmc will check `language_filest::lazy_method_map` to determine this, test-gen will always answer yes). That will remove the need for annotation on stub creation as is happening in this PR.

The provide-stub-body callback is an entirely separate matter, and should be pursued independently.